Framework/Crystal/src/PredictPeaks.cpp

```diff
@@ -489,10 +492,58 @@ void PredictPeaks::calculateQAndAddToOutput(const V3D &hkl,
   if (m_sfCalculator) {
     p.setIntensity(m_sfCalculator->getFSquared(hkl));
   }
-
+  int edge = this->getProperty("EdgePixels");
+  if (edge > 0) {
+    if (edgePixel(p.getBankName(), p.getCol(), p.getRow(), edge))
+      return;
+  }
   // Add it to the workspace
   m_pw->addPeak(p);
 }
-
+//-----------------------------------------------------------------------------------------
+/**
+  @param  bankName     Name of bank containing peak
+  @param  col          Column number containing peak
+  @param  row          Row number containing peak
+  @param  Edge         Number of edge points for each bank
+  @return True if peak is on edge
+*/
+bool PredictPeaks::edgePixel(std::string bankName, int col, int row, int Edge) {
```

Comment:
I did a search for `edgePixel` in the code base and as far as I can see this is the exact same function as in `SCDCalibratePanels.cpp`. `CentroidPeaks` and `OptimizeLatticeForCellType` contain a slightly different version (it could be just explicit types instead of auto, I have not checked in detail).

The fact that this comes up in four different places suggests that it might be a more general concept that should probably be implemented once in some location where it can be used by all these four algorithms.

I haven't thought about it in detail, but would it be an option to mask the edge pixels and simply check whether a detector is masked? I'm not sure about the best way to implement this masking though. Having an algorithm `MaskEdgePixels` would be one option, it could create a proper mask given some instrument. Can such a mask be reverted easily?